src/pipdeptree/_models/package.py

```diff
@@ -241,9 +280,13 @@ def is_conflicting(self) -> bool:
         # unknown installed version is also considered conflicting
         if self.installed_version == self.UNKNOWN_VERSION:
             return True
+
         ver_spec = self.version_spec if self.version_spec else ""
-        req_version_str = f"{self.project_name}{ver_spec}"
-        req_obj = Requirement.parse(req_version_str)  # type: ignore[no-untyped-call]
+        if ver_spec:
+            req_obj = SpecifierSet(ver_spec)
+        else:
+            return False
+
         return self.installed_version not in req_obj
```

Comment:
We can do something like:

```suggestion
        return self.installed_version not in SpecifierSet(ver_spec) if ver_spec else False
```